### PR TITLE
update sizing guide with user event metrics info

### LINKED ISSUE #35640: Update the sizing guide with an indicator on which user events to use
### Description

 Update the sizing guide with an indicator on which user events to use to measure logins, refreshes, and client client credential grants.

### Discussion

https://github.com/keycloak/keycloak/issues/34503

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by kami619**
fixes #35640

### CODE REVIEW

**Comment by ahus1** [comment]
@kami619 - thank you for this change. Pointing to the user metrics is a good start. 

I would like it to be more specific about the events that are relevant to the sizing guide, so you would name `login`, `refresh_token` and `client_login` (not sure about the last one, please verify). 

I see that it is currently part of the "Calculation example" section. If we would move it to its own section, we can add more context on when and how to use this. A draft statement in that new section could be "Use the the metrics X, Y and Z to measure the activity around user login, refresh token and client credential grant to validate your inputs for sizing calculation, monitor user activity load changes in the daily and weekly cycle, and watch out for trends that would require re-sizing your system". 

If we see that the `login` event captures not only password based logins but also cookie based logins, this could be a note in the documentation as well. Maybe using some one the HTTP metrics can be a help to sort cookie based logins from password based logins: Password based logins will show as s POST once you submit the form, and the cookie based ones won't. 

Please expand and add more context and details as needed :D

**Comment by kami619** [comment]
@ahus1  thanks for the feedback. I have updated the docs, but still couple of things are pending from how I want it to look, such as verifying the metrics for particular events. 

You could still share me any more feedback based on the changes I did so far.

File: docs/guides/high-availability/concepts-memory-and-cpu-sizing.adoc [hunk 1]
```diff
@@ -61,6 +61,62 @@ For every 100 login/logout/refresh requests per second:
...
+`client_login`  to monitor and analyze activities related to user logins,
+refresh token requests, and client credential grants respectively.
+
+These metrics are crucial for tracking daily and weekly fluctuations in user activity loads,
+identifying emerging trends that may indicate the need to resize the system and
+validating sizing calculations.
+By systematically measuring and evaluating these user event metrics,
+you can ensure your system remains appropriately scaled and responsive to changes in user behavior and demand.
+
+This is a preview feature and can be enabled with a feature flag.
```
Comment (ahus1) [inline docs/guides/high-availability/concepts-memory-and-cpu-sizing.adoc:76]: I suppose we already put that in the main docs about this one, and I wouldn't name it here again, as would need to update all of these if it is no longer preview.

```suggestion
```

File: docs/guides/high-availability/concepts-memory-and-cpu-sizing.adoc [hunk 2]
```diff
@@ -61,6 +61,62 @@ For every 100 login/logout/refresh requests per second:
...
+
+This section details the use of the user event metrics such as `login` , `refresh_token` ,
+`client_login`  to monitor and analyze activities related to user logins,
+refresh token requests, and client credential grants respectively.
+
+These metrics are crucial for tracking daily and weekly fluctuations in user activity loads,
+identifying emerging trends that may indicate the need to resize the system and
+validating sizing calculations.
+By systematically measuring and evaluating these user event metrics,
+you can ensure your system remains appropriately scaled and responsive to changes in user behavior and demand.
```
Comment (ahus1) [inline docs/guides/high-availability/concepts-memory-and-cpu-sizing.adoc:74]: This is showing-off maybe a bit too much. 

I want to make a point here that you can use these three events as inputs for your sizing calculation, and I think this get a bit lost.

File: docs/guides/high-availability/concepts-memory-and-cpu-sizing.adoc [hunk 3]
```diff
@@ -61,6 +61,62 @@ For every 100 login/logout/refresh requests per second:
...
+
+This is a preview feature and can be enabled with a feature flag.
+See the link:{links_observability_event-metrics_url}[{observability_event-metrics_name}] for more information.
+
+==== PromQL queries:
+
```
Comment (ahus1) [inline docs/guides/high-availability/concepts-memory-and-cpu-sizing.adoc:80]: Adding the PromQL queries to a sizing guide seems to be a bit too much here as it is not about monitoring, but sizing. Maybe we would eventually add a section to https://www.keycloak.org/nightly/observability/metrics-for-troubleshooting once we decide to add some of these events to the dashboard.

**Review by ahus1 — COMMENTED** [review]
Thank you for this initial draft, and for verifying the events. 
See below for some nitpicks and comments.

File: docs/guides/high-availability/concepts-memory-and-cpu-sizing.adoc (hunk 2 above)
Comment (kami619) [reply docs/guides/high-availability/concepts-memory-and-cpu-sizing.adoc:74]: I would argue, its important to highlight the possibility of what could be achieved, but I don't want to compromise the original intent.

File: docs/guides/high-availability/concepts-memory-and-cpu-sizing.adoc (hunk 3 above)
Comment (kami619) [reply docs/guides/high-availability/concepts-memory-and-cpu-sizing.adoc:79]: fair enough. I will remove this section and keep it safe for a future commit.

**Review by ahus1 — APPROVED** [review]
Thank you for the PR. I've updated the heading to capture more of the user's need, and added an introductory sentence. 

The goal the overarching epic is that Keycloak can provide the necessary inputs to the sizing guide using metrics. With the password based login, we currently fall short here, which is a pity. I've added the HTTP server metrics as a substitution.

Let's discuss in the team how we can add a better metric here that allows for simpler sizing.

**Pull request merged**